infra/gravity/terraform.go

```diff
@@ -146,8 +148,105 @@ func saveResourceAllocations() error {
 	return nil
 }
 
+// makeDynamicParams takes base config, validates it and returns cloudDynamicParams
+func makeDynamicParams(baseConfig ProvisionerConfig) (*cloudDynamicParams, error) {
+	param := cloudDynamicParams{ProvisionerConfig: baseConfig}
+
+	// OS name is cloud-init script specific
+	// enforce compatible values
+	var ok bool
+	usernames := map[string]map[string]string{
+		"azure": map[string]string{
+			"ubuntu": "robotest",
+			"debian": "admin",
+			"redhat": "redhat", // TODO: check
+			"centos": "centos",
+			"suse":   "robotest",
+		},
+		"aws": map[string]string{
+			"ubuntu": "ubuntu",
+			"debian": "admin",
+			"redhat": "redhat",
+			"centos": "centos",
+		},
+	}
+
+	param.user, ok = usernames[baseConfig.CloudProvider][baseConfig.os]
+	if !ok {
+		return nil, trace.BadParameter("%s ")
+	}
+
+	param.homeDir = filepath.Join("/home", param.user)
```

Comment:
The location of the home directory can be set anywhere. Maybe 
```shell
getent passwd $user
```
would be safer?